Packit service generated version make tar fail

Hello,

I've found this issue when I was adding Packit support to one of our projects. Seems that tar has an issue with generated number for the package version from Packit.

https://prod.packit.dev/srpm-build/9864/logs

```
Message: Preparing of the upstream to the SRPM build failed: Command failed (rc=2, reason={"metadata":{},"status":"Failure","message":"command terminated with non-zero exit code: Error executing in Docker Container: 2","reason":"NonZeroExitCode","details":{"causes":[{"reason":"ExitCode","message":"2"}]}})
TEMP_DIR=$(mktemp --tmpdir -d initial-setup-localization-XXXXXXXXXX) && \
git clone --depth 1 -b master -- https://github.com/rhinstaller/initial-setup-l10n.git $TEMP_DIR && \
cp $TEMP_DIR/master/*.po ./po/ && \
rm -rf $TEMP_DIR
Cloning into '/tmp/initial-setup-localization-Ekg2Mdt3Yx'...
make[1]: Entering directory '/sandcastle/docker-io-usercont-sandcastle-prod-20201116-122742104903'
(GIT_DIR=.git git log > .changelog.tmp && mv .changelog.tmp ChangeLog; rm -f .changelog.tmp) || (touch ChangeLog; echo 'git directory not found: installing possibly empty changelog.' >&2)
make[1]: Leaving directory '/sandcastle/docker-io-usercont-sandcastle-prod-20201116-122742104903'
git archive --format=tar --prefix=initial-setup-0.3.84/ r0.3.84-1 > initial-setup-0.3.84.tar
mkdir initial-setup-0.3.84
cp -r po initial-setup-0.3.84
cp ChangeLog initial-setup-0.3.84/
tar -rf initial-setup-0.3.84.tar initial-setup-0.3.84
tar: value 1021440000 out of uid_t range 0..2097151
tar: Exiting with failure status due to previous errors
make: *** [Makefile:81: archive] Error 2
```

From my understanding tar archive couldn't be modified because of name but it's possible that I understand it wrong. It's not happening when the SRPM archive is created locally.

https://github.com/rhinstaller/initial-setup/blob/2ebb4ba4c2493d4a5fc8b61e9e872044ef383169/Makefile#L71

Not a priority issue for us. Instead of fixing this I'm changing how the project is packaged but it could be a valid issue for another project.

--- a/content/docs/actions.md
+++ b/content/docs/actions.md
@@ -75,9 +75,9 @@ name. The best practice is to do it from the last step and print it: `bash -c
 
 ### `fix-spec-file`
 
-By default, this action updates the specfile so it's possible to have the spec
-reference the content of the archive and unpack it during the build properly.
-The action tries to perform 3 operations on a spec file:
+By default, this action updates the spec file so it's possible to have a proper
+reference of the archive in the `%prep` section and unpack it during the build
+properly. The action tries to perform 3 operations on a spec file:
 
 1. It replaces Source configured by [`spec_source_id`](/docs/configuration/#spec_source_id) (default `Source0`) with a local path to the generated tarball
 2. It changes the first `%setup` (or `%autosetup`) macro in `%prep` and adds `-n` so the generated

--- a/content/docs/actions.md
+++ b/content/docs/actions.md
@@ -26,7 +26,9 @@ e.g. cloning an upstream repo.
 
 Currently, these are the actions you can use:
 
-## `propose-downstream` command
+## Command matrix
+
+### `propose-downstream` command
 
 |        | name                  | working directory | when run                                                                          | description                               |
 | ------ | --------------------- | ----------------- | --------------------------------------------------------------------------------  | ----------------------------------------- |
@@ -37,7 +39,7 @@ Currently, these are the actions you can use:
 |        | `get-current-version` | upstream git repo | when the current version needs to be found                                        | expect version as a stdout parameter      |
 
 
-## Creating SRPM
+### Creating SRPM
 
 These apply to the `srpm` command and building in COPR.
 
@@ -49,16 +51,41 @@ These apply to the `srpm` command and building in COPR.
 |        | `create-patches`      | upstream git repo | after sync of upstream files to the downstream                                    | replace patching                          |
 |        | `fix-spec-file`            | upstream git repo | after creation of a tarball and before running rpmbuild command                   | this action changes spec file to use the new tarball                          |
 
-**create-archive** - is expected to return a relative path to the archive - relative within the repository. If there are more steps, then one of them has to return the archive name.
+## Actions details
 
-**fix-spec-file** — this action updates the specfile so it's possible to have the spec reference
-                    the tarball and unpack it. This method tries to perform 3 operations on a spec file:
+All actions are executed in a locked-down OpenShift pod. Your commands are
+invoked with arbitrary UIDs from a high range. Some tools may experience
+problems with these UIDs, such as `tar`. You can observe an error like this:
+```
+tar: value 1021440000 out of uid_t range 0..2097151
+tar: Exiting with failure status due to previous errors
+```
+
+For `tar`, it's [recommended](https://github.com/packit/packit.dev/issues/192#issuecomment-729594207)
+to use the `pax` format (`tar -H pax`).
+
+If you run into similar issues with other tools, please consult documentation or maintainers of the project.
+
+### `create-archive`
+
+It is expected to return a relative path within the repository to the generated
+archive. If there are more steps, then one of them has to return the archive
+name. The best practice is to do it from the last step and print it: `bash -c
+'echo path/to/archive-$VERSION.tar.gz'`.
+
+### `fix-spec-file`
+
+By default, this action updates the specfile so it's possible to have the spec
+reference the content of the archive and unpack it during the build properly.
+The action tries to perform 3 operations on a spec file:
 
 1. It replaces Source configured by [`spec_source_id`](/docs/configuration/#spec_source_id) (default `Source0`) with a local path to the generated tarball
 2. It changes the first `%setup` (or `%autosetup`) macro in `%prep` and adds `-n` so the generated
  tarball can be unpacked (it tries to extract the directory name directly from the archive
  or uses the configured [`archive_root_dir_template`](/docs/configuration#archive_root_dir_template))
 3. It changes %version
+
+If you provide your own implementation, none of the above happens.
 
 For example a package may define multiple Sources. In such cases, the
 default implementation of `fix-spec-file` won't be able to update `%prep`

